regression_tests/actor_service.go

```diff
@@ -0,0 +1,95 @@
+package regressiontests
+
+import (
+	"context"
+	"fmt"
+
+	wallet "github.com/bitcoin-sv/spv-wallet-go-client"
+	"github.com/bitcoin-sv/spv-wallet-go-client/config"
+	"github.com/bitcoin-sv/spv-wallet-go-client/walletkeys"
+	"github.com/bitcoin-sv/spv-wallet/models/response"
+)
+
+// actor represents an individual user or entity within the SPV Wallet ecosystem.
+// It includes details like the alias, private key (xPriv), public key (xPub), and paymail address.
+type actor struct {
+	alias   string // The unique alias for the actor.
+	xPriv   string // The extended private key for the actor.
+	xPub    string // The extended public key for the actor.
+	paymail string // The paymail address associated with the actor.
+}
+
+// newActor creates and returns a new Actor instance with a unique alias and domain.
+// It generates a new set of random wallet keys (xPriv and xPub) and constructs the paymail address.
+// On success, it returns the initialized Actor instance and a nil error.
+// If key generation fails, it returns a non-nil error.
+func newActor(alias, domain string) (*actor, error) {
+	keys, err := walletkeys.RandomKeys()
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate wallet keys: %w", err)
+	}
+	return &actor{
+		alias:   alias,
+		xPriv:   keys.XPriv(),
+		xPub:    keys.XPub(),
+		paymail: alias + "@" + domain,
+	}, nil
+}
+
+// actorService provides an interface to interact with the SPV Wallet API.
+// It supports operations such as retrieving the balance, accessing the paymail,
+// fetching transactions, and transferring funds within the wallet ecosystem.
+type actorService struct {
+	userAPI         *wallet.UserAPI  // The API client for interacting with the SPV Wallet.
```

Comment:
@mgosek-4chain please don't be angry, but in general while I like the architecture of the go-client itself, at least for me when it comes to regression-testing there are too many abstractions here, services thrown in.
It seems to me that the code loses a bit of readability by being so scattered and fragmented.
This is absolutely not a blocker to the merge, but that's how I wanted to share my opinion :D